base_rest_pydantic/restapi.py

```diff
@@ -154,10 +159,23 @@ def from_params(self, service, params):
 
     def to_response(self, service, result):
         self._do_validate(result, "output")
-        return [
+        json_res = {}
+        json_res["metadata"] = self.to_metadata_response(service, result["metadata"])
+        json_res["data"] = [
```

Comment:
Why do you modify the structure of the response? We don't want to always return "metadata" and "data". Even if it's a list. We must keep the generic behavior. If you need such an approach for your services you are free to develop your own rest parameter class. Do you've try to use the generic types to define a generic model to encapsulate your response?

```python
from typing import Any, TypeVar, Generic, List

_T = TypeVar("_T", bound=Any)

from extendable_pydantic import ExtendableModelMeta

from pydantic import BaseModel


class SearchResponse(BaseModel,  Generic[_T], metaclass=ExtendableModelMeta):
    size: int = None
    data: List[_T]


 # to use it:
class Partner(BaseModel, metaclass=ExtendableModelMeta):
    name: str

results = SearchResponse[Partner](data=[Partner(name=1)], size=1)

.... 
```

Reply:
@PierrickBrun If fixed the use of generics with pydantic and extendable... see https://github.com/lmignon/extendable-pydantic/pull/1 (At this stage it doesn't work with py < 3.8)